onnx/defs/object_detection/defs.cc

```diff
@@ -0,0 +1,109 @@
+    OpSchema()
+      .SetDoc(RoIAlign_ver1_doc)
+      .Attr(
+          "spatial_scale",
+          "Multiplicative spatial scale factor to translate ROI coordinates "
+          "from their input spatial scale to the scale used when pooling, "
+          "i.e., spatial scale of the input feature map X relative to the "
+          "input image. E.g.; default is 1.0f. ",
+          AttributeProto::FLOAT,
+          1.f)
+      .Attr(
+          "output_height",
+          "default 1; Pooled output Y's height.",
+          AttributeProto::INT,
+          static_cast<int64_t>(1))
+      .Attr(
+          "output_width",
+          "default 1; Pooled output Y's width.",
+          AttributeProto::INT,
+          static_cast<int64_t>(1))
+      .Attr(
+          "sampling_ratio",
+          "Number of sampling points in the interpolation grid used to compute "
+          "the output value of each pooled output bin. If > 0, then exactly "
+          "sampling_ratio x sampling_ratio grid points are used. If == 0, then "
+          "an adaptive number of grid points are used (computed as "
+          "ceil(roi_width / output_width), and likewise for height). Default is 0.",
+          AttributeProto::INT,
+          static_cast<int64_t>(0))
+      .Attr(
+        "mode",
+        "The pooling method. Two modes are supported: 'avg' and 'max'. "
+        "Default is 'avg'.",
+        AttributeProto::STRING,
+        std::string("avg"))
+      .Input(
+          0,
+          "X",
+          "Input data tensor from the previous operator; "
+          "4-D feature map of shape (N, C, H, W) "
+          "where N is the batch size, C is the number of channels, "
+          "and H and W are the height and the width of the data.",
+          "T1")
+      .Input(
+          1,
+          "rois",
+          "RoIs (Regions of Interest) to pool over; rois is "
+          "2-D input of shape (num_rois, 5) given as "
+          "[[batch_idx, x1, y1, x2, y2], ...]. "
+          "The RoIs' coordinates are in the coordinate system of the input image.",
+          "T1")
+      .Output(
+          0,
+          "Y",
+          "RoI pooled output, 4-D tensor of shape "
+          "(num_rois, C, output_height, output_width). The r-th batch element Y[r-1] "
+          "is a pooled feature map corresponding to the r-th RoI X[r-1].",
+          "T1")
+      .TypeConstraint(
+          "T1",
+          {"tensor(float16)", "tensor(float)", "tensor(double)"},
+          "Constrain types to float tensors.")
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+          if (!hasNInputShapes(ctx, 3)) {
+            return;
+          }
+          propagateElemTypeFromInputToOutput(ctx, 0, 0);
+
+          auto& input_shape = getInputShape(ctx, 0);
+          auto& rois_shape = getInputShape(ctx, 1);
+          auto* output_shape = getOutputShape(ctx, 0);
+
+          output_shape->clear_dim();
+          output_shape->add_dim()->set_dim_value(static_cast<int64_t>(
```

Comment:
Minimum check/assert is recommended before accessing the shape. It will help us detect the problem early and print out reasonable error message.